Step1

ivvve opened the pull request:
Step1 구현하였습니다.
많이 부족하지만 잘 부탁드리겠습니다 감사합니다!

ivvve committed:
refactor: changed argument of 'move' method to use MovingStrategy interface

ivvve committed:
test: added Car unit tests

ivvve committed:
test: added test codes when the parameter is "null" or "empty string"

ivvve committed:
chore: deleted empty.txt

ivvve committed:
feature: added feature

- when expression is empty or null then return 0

ivvve committed:
feature: added VO stands for positive number

ivvve committed:
feature: added calculate feature using default separators

ivvve committed:
feature: added calculate using custom separator

ivvve committed:
feature: added validation when custom separation declaration format is wrong

ivvve committed:
refactor: changed removing custom separator logic

ivvve committed:
commit gradle-wrapper changed properties

ivvve committed:
Merge remote-tracking branch 'upstream/ivvve' into ivvve

pangyocat commented on src/main/java/camp/nextstep/edu/calculator/PositiveNumber.java:
```diff
@@ -0,0 +1,33 @@
+
+    boolean isSameValue(final int value) {
+        return (this.value == value);
+    }
```
테스트 용도로만 쓰는 것 같은데 equals를 재정의하는건 어떨까요?
```
 assertThat(numbers.get(0)).isEqualTo(new PositiveNumber(1));
```

pangyocat commented on src/main/java/camp/nextstep/edu/calculator/PositiveNumber.java:
```diff
@@ -0,0 +1,33 @@
+        this.value = value;
+    }
+
+    int getValue() {
```
이름 하나로 차후 double, float 타입도 가능하게 확장성도 높이고 어떤 타입의 숫자 value가 넘어오는지 명확히 알 수 있게 toInt, intValue로 변경이 가능 할 것 같습니다!
가장 좋은 예로 Integer가 있죠 😄

pangyocat commented on src/main/java/camp/nextstep/edu/calculator/Expression.java:
```diff
@@ -0,0 +1,83 @@
+
+        final String[] numbers = expressionWithoutCustomSeparator.split(DEFAULT_SEPARATORS);
+        return Arrays.stream(numbers)
+                .map(number -> new PositiveNumber(Integer.parseInt(number)))
```
```suggestion
                .map(Integer::parseInt)
                .map(PositiveNumber::new)
```

pangyocat commented on src/main/java/camp/nextstep/edu/calculator/Expression.java:
```diff
@@ -0,0 +1,83 @@
+
+    List<PositiveNumber> retrieveNumbers() {
+        final String expressionWithoutCustomSeparator = this.hasCustomSeparator() ?
+                this.getExpressionWithoutCustomSeparator() : this.expression;
```
커스텀 구분자 여부에 따른 처리를 `getExpressionWithoutCustomSeparator`로 옮기면 어떨까요?

ivvve replied:
제가 구현한 로직(커스텀 구분자를 디폴트 구분자로 치환)같은 경우
`getExpressionWithoutCustomSeparator`에 이 삼항연산자 로직을 하나로 빼는 편이 더 의미있는 것 같습니다.
꼼꼼한 확인 감사합니다 👍

pangyocat commented on src/main/java/camp/nextstep/edu/calculator/StringAdder.java:
```diff
@@ -0,0 +1,25 @@
+    private static PositiveNumber getSumOf(final List<PositiveNumber> numbers) {
+        return numbers.stream()
+                .reduce(PositiveNumber::add)
+                .orElseGet(() -> new PositiveNumber(0));
```
`orElseGet` 👍

pangyocat reviewed (approved):
테스트 코드 작성과 객체간 분리도 그렇고 깔끔히 잘 작성해 주셨네요! 예외에 모두 메시지를 추가하신 꼼꼼함까지! 💯
커스텀 구분자를 디폴트 구분자로 치환하여 처리하는 방법은 생각 못해봤었는데 좋은 방법이네요.

추가적으로 디폴트 접근제한자 사용하신 부분들이 보이던데 의도하신 부분이라면 👍
간단한 피드백 남겨드렸는데 스타일의 차이 정도인 부분도 있어서 원하신다면 마무리하셔도 좋을 것 같습니다 😄

ivvve committed:
refactor: replaced 'isSameValue' method to 'equals'

ivvve committed:
refactor: changed method name from 'getValue' to 'toInt'

(it can specify what type is in this VO)

ivvve committed:
refactor: refactored Expression's logic

- moved ternary operation logic into 'getExpressionWithoutCustomSeparator' method
- split lambda expression
